### DISCUSSION TITLE
Customising theme using css variables

### DISCUSSION THREAD

**Discussion by viktorromanyuk** [element-plus/element-plus#12203] [Q&A] [answered]
Hi there! I'm trying to customize theme using css variables but it doesn't work as expected. When I set 
```
:root {
  --el-color-primary: green;
}
```
the main color works fine, however by hovering for example on button, the bg color remains as default blue as it expects to have --el-color-primary-light-3 which is not changed. So should I define all the colors for `--el-color-primary`, `--el-color-primary-light-3`, etc. variables to make it properly work or I'm missing somethig?

Overriding colors using scss @forward works perfectly fine but I need that thing with css variables. 

Will appreciate any help!

**Comment by gimjin** [comment, marked as answer]
refer [The Future of CSS Color Functions](https://una.im/css-color-theming/), css doesn't mix colors well. so you should define others css variables.

Cover variables with scss, there will also be css variables at runtime.

**Reply by viktorromanyuk** [reply]
That's what I initially thought. Thanks anyways!

**Comment by gimjin** [comment]
Hi @viktorromanyuk [Color Mix postcss plugin](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-color-mix-function) might make your code prettier. Now [color-mix](https://cssdb.org/#color-mix) function in stage 2.
